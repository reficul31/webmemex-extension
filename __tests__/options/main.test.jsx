import React from 'react'
import Layout from 'src/options/layout'
import Routes from 'src/options/routes'
import { Router, Route, IndexRedirect, hashHistory } from 'react-router'
import renderer from 'react-test-renderer'

describe('Main component tests', () => {
    test('should mount without error and test against the given snapshot', () => {
        const component = renderer.create(
            <Router history={hashHistory}>
                <Route path='/' component={Layout}>
                    <IndexRedirect to='/settings' />
                    { Routes.map(route =>
                        <Route
                            key={route.pathname}
                            path={route.pathname}
                            component={route.component}
                    />
                )}
                </Route>
            </Router>
            )
        const tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })
})
