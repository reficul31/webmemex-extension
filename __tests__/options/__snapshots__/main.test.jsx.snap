// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main component tests should mount without error and test against the given snapshot 1`] = `
<div
  className="root"
>
  <nav
    className="root"
  >
    <h1
      className="title"
    >
      Web Memex
    </h1>
    <ul
      className="nav"
    >
      <li
        className="navItem"
      >
        <a
          className="navLink"
          href="#/import"
          onClick={[Function]}
          style={Object {}}
        >
          Import
        </a>
      </li>
      <li
        className="navItem"
      >
        <a
          className="navLink isActive"
          href="#/settings"
          onClick={[Function]}
          style={Object {}}
        >
          Settings
        </a>
      </li>
    </ul>
  </nav>
  <div
    className="route"
  >
    <div>
      <h1
        className="routeTitle"
      >
        Settings
      </h1>
      <p>
        Settings specific components to come in here.
      </p>
    </div>
  </div>
</div>
`;
